lib/spack/spack/solver/asp.py

```diff
@@ -1411,8 +1476,7 @@ def condition(
         imposed_spec: Optional[spack.spec.Spec] = None,
         name: Optional[str] = None,
         msg: Optional[str] = None,
-        transform_required: Optional[TransformFunction] = None,
-        transform_imposed: Optional[TransformFunction] = remove_node,
+        context: Optional[ConditionContext] = None,
```

Comment:
The docstring needs an update in this function

Reply:
I've updated docstrings (and also added some to the `*Context` objects.